File: Java/LinkedList/RemoveNthFromEnd.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     public int val;
 *     public ListNode next;
 *     ListNode(int x) { val = x; next = null; }
 * }
 */
public class Solution {
	public ListNode removeNthFromEnd(ListNode a, int b) {
	    if(a == null)return null;
	    ListNode pt1 = a, pt2 = a;
	    for(int i = 0; i <= b; i++){
	        if(pt2 != null)
	            pt2 = pt2.next;
	       else return a.next;
	    }

	    while(pt2 != null){
	        pt1 = pt1.next;
	        pt2 = pt2.next;
	    }
	   pt1.next = pt1.next.next;

	   return a;
	}
}

File: JS/LinkedList/RemoveNthFromEnd.js
// Definition for singly-linked list.
//			function Node(data){
//				this.data = data
//				this.next = null
//			}

module.exports = {
	//param A : head node of linked list
	//param B : integer
	//return the head node in the linked list
	removeNthFromEnd : function(list, n){
        var l1 = list, l2 = list;
        for(var i = 0; i <= n; i++){
            if(l1 != null)
                l1 = l1.next;
            else return list.next;
        }
        while(l1 != null){
            l1 = l1.next;
            l2 = l2.next;
        }
        l2.next = l2.next.next;
        return list;
	}
};
